web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -554,9 +594,19 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
               // and a loading indicator if needed.
               !unifiedResourcePreferencesAttempt ||
               hasFinished(unifiedResourcePreferencesAttempt)
-                ? resources.map(unifiedResource => ({
-                    item: mapResourceToViewItem(unifiedResource),
-                    key: generateUnifiedResourceKey(unifiedResource.resource),
+                ? resources.map(({ ui, resource }) => ({
+                    item: mapResourceToViewItem({
+                      ui,
+                      resource: {
+                        ...resource,
+                        // if we are in 'requestable' only mode, then all resources returned
+                        // require a request and should be displayed that way
+                        requiresRequest:
+                          resource.requiresRequest ||
+                          availabilityFilter?.mode === 'requestable',
```

Comment:
I think I asked about this, but I'm still not sure why do we need it.

Reply:
Also, it looks like we don't set `requiresRequest` for dbs? 

https://github.com/gravitational/teleport/assets/20583051/df656e42-3f97-49fb-a42e-1a9bf7b701ab